server/models/listProductModel.js

```diff
@@ -3,27 +3,38 @@ const UserListModel = require("./userListModel");
 const ProductModel = require("./productModel");
 const db = require("./index");
 
-const ListProductModel = db.define("ListProduct", {
-  id: {
-    type: dt.INTEGER,
-    primaryKey: true,
-    autoIncrement: true,
-  },
-});
+const ListProductModel = db.define("ListProduct", {});
 
-UserListModel.hasMany(ListProductModel, {
+UserListModel.belongsToMany(ProductModel, {
+  through: ListProductModel,
   foreignKey: {
-    name: "list_id",
+    name: "listId",
     allowNull: false,
   },
   onDelete: "CASCADE",
 });
-ProductModel.hasMany(ListProductModel, {
+ProductModel.belongsToMany(UserListModel, {
+  through: ListProductModel,
   foreignKey: {
-    name: "product_id",
+    name: "productId",
     allowNull: false,
   },
   onDelete: "CASCADE",
 });
 
+// UserListModel.hasMany(ListProductModel, {
+//   foreignKey: {
+//     name: "listId",
+//     allowNull: false,
+//   },
+//   onDelete: "CASCADE",
+// });
+// ProductModel.hasMany(ListProductModel, {
+//   foreignKey: {
+//     name: "productId",
+//     allowNull: false,
+//   },
+//   onDelete: "CASCADE",
+// });
```

Comment:
Will this be transferred onto a different file? 

Or maybe this was just commented out for testing purposes?

Reply:
oh yea, that needs to get removed!

Reply:
Arent the listId and productId used to find which products are in a list?

Reply:
Yes. I'm using a different association method called `belongsToMany` to set up those columns.